### PR TITLE
feat: read config from file

### LINKED ISSUE #382: Load configuration from file
To support self hosted users better, and to make sure we don't pollute the environment variables for the Docker Compose file, we really should provide loading configuration from file, while still supporting reading it directly from the environment variables.

One thing why this should be done is to provide better visibility for people who wants to set their blob storage to anything other than Docker volume (ex. gcs, s3, azblob)

A few examples:
- Relay: https://github.com/getsentry/self-hosted/blob/master/relay/config.example.yml
- Sentry: https://github.com/getsentry/self-hosted/blob/master/sentry/config.example.yml / https://github.com/getsentry/self-hosted/blob/master/sentry/sentry.conf.example.py
- Symbolicator: https://github.com/getsentry/self-hosted/blob/master/symbolicator/config.example.yml

### PR DESCRIPTION
**Pull request by aldy505**
Closes #382 

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

**Comment by aldy505** [comment]
@phacops @viglia any updates?

**Comment by williamdes** [comment]
Maybe @bmckerry knows where to your this PR for review?

**Comment by stayallive** [comment]
This would be very nice to see, it would make it very simple for self-hosted to for example store profiles in bucket storage instead of on disk!

**Comment by aldy505** [comment]
another gentle ping @phacops @viglia

**Comment by phacops** [comment]
Hi @aldy505,

Sorry for the delay! It seems like this PR will now require us to ship a config file while still being able to override it via environment variables. Why is this needed versus shipping a `.env` file to be loaded?

**Comment by aldy505** [comment]
@phacops It's much easier to do configuration by using file instead of environment variables for self-hosted. Also the current state of self-hosted configuration services are using configuration files instead of just bare environment variable (although environment variables are still in place).

- Relay: https://github.com/getsentry/self-hosted/blob/master/relay/config.example.yml
- Sentry: https://github.com/getsentry/self-hosted/blob/master/sentry/config.example.yml and https://github.com/getsentry/self-hosted/blob/master/sentry/sentry.conf.example.py
- Symbolicator: https://github.com/getsentry/self-hosted/blob/master/symbolicator/config.example.yml

Modifying `docker-compose.yml` file directly on the self-hosted repo to include the environment variables is somewhat discouraged because it's easier for the upgrade process to just do `git checkout <VERSION>` rather than doing a `git diff` to the git tag and apply the changes on your own.

**Comment by phacops** [comment]
I don't really see what's easier with a separate file.

We still have to load the configuration via Docker Compose (like https://github.com/getsentry/self-hosted/blob/master/docker-compose.yml#L309-L314) or on Kubernetes.

What I could do to make it more compatible is to source the `.env` file before we launch `vroom`. That way, it's compatible with every container platform and we still have to load the file in the container except, with Docker compose, it'd be a single line.

As for the configuration documentation, is reading a YAML file so much better than a Go struct?

The only argument in the favor is doing this is "other services are doing it that way" but everything else seems more complicated and less integrated with existing tools.

File: cmd/vroom/main.go [hunk 1]
```diff
@@ -50,9 +52,16 @@ const (
 
 func newEnvironment() (*environment, error) {
 	var e environment
-	err := cleanenv.ReadEnv(&e.config)
+	err := cleanenv.ReadConfig("config.yml", &e.config)
```
Comment (phacops) [inline cmd/vroom/main.go:55]: I would prefer not to hardcode the filename and make it an argument like other services.

**Comment by hubertdeng123** [comment]
Chiming in here, the reason why symbolicator and relay have these yml config files is due to a design choice for those services. If @phacops believes that this is not the best approach for configuring vroom and prefers loading in variables from the env, I don't think we should go forwards with using a yml config file for customizing self-hosted. Having said that, the proposed `env_file` solution requires docker compose v2.24.0+. Perhaps until we can enforce a min requirement of using docker compose v2.24.0+, we can just add the env variables needed for vroom in the .env file.

File: cmd/vroom/config.go [hunk 2]
```diff
@@ -2,20 +2,30 @@ package main
...
 
-		SnubaHost string `env:"SENTRY_SNUBA_HOST" env-default:"http://localhost:1218"`
+		SnubaHost string `env:"SENTRY_SNUBA_HOST" yaml:"snuba_host" env-default:"http://localhost:1218"`
 
-		BucketURL string `env:"SENTRY_BUCKET_PROFILES" env-default:"file://./test/gcs/sentry-profiles"`
+		BucketURL string `env:"SENTRY_BUCKET_PROFILES" yaml:"bucket_url" env-default:"file://./test/gcs/sentry-profiles"`
+
+		Logging struct {
+			Level  string `env:"SENTRY_LOGGING_LEVEL" yaml:"level" env-default:"info"`
+			Format string `env:"SENTRY_LOGGING_FORMAT" yaml:"format" env-default:"simplified"`
```
[the remark this answers is not in the document]
Comment (aldy505) [reply cmd/vroom/config.go:28]: It can be `json` (default) or `simplified` (for human readable)

**Comment by phacops** [comment]
We're not going to implement reading configuration from files.

**Pull request closed without merging**